src/js/tabs/setup.js

```diff
@@ -364,80 +364,104 @@ setup.initialize = function (callback) {
+            if (buildKeyValid && isIspConnected) {
+                const buildRoot = getBuildRootBaseUri();
+
+                // Creates the "Config" button
                 const buildConfig = `<span class="buildInfoBtn" title="${i18n.getMessage(
                     "initialSetupInfoBuildConfig",
                 )}: ${buildRoot}/json">
-                                    <a href="${buildRoot}/json" target="_blank"><strong>${i18n.getMessage(
+                    <a href="${buildRoot}/json" target="_blank"><strong>${i18n.getMessage(
     "initialSetupInfoBuildConfig",
 )}</strong></a></span>`;
+
+                // Creates the "Log" button
                 const buildLog = `<span class="buildInfoBtn" title="${i18n.getMessage(
                     "initialSetupInfoBuildLog",
                 )}: ${buildRoot}/log">
-                                    <a href="${buildRoot}/log" target="_blank"><strong>${i18n.getMessage(
+                    <a href="${buildRoot}/log" target="_blank"><strong>${i18n.getMessage(
     "initialSetupInfoBuildLog",
 )}</strong></a></span>`;
-                build_info_e.html(`<span class="buildInfoBtn" title="${i18n.getMessage(
-                    "initialSetupInfoBuildOptions",
-                )}">
-                    <a class="buildOptions" href=#"><strong>${i18n.getMessage(
-        "initialSetupInfoBuildOptionList",
-    )}</strong></a></span>`);
+
+                // Shows the "Config" and "Log" buttons
                 build_info_e.html(`${buildConfig} ${buildLog}`);
             } else {
                 build_info_e.html(
-                    supported ? i18n.getMessage("initialSetupNotOnline") : i18n.getMessage("initialSetupNoBuildInfo"),
+                    isIspConnected
+                        ? i18n.getMessage("initialSetupNoBuildInfo")
+                        : i18n.getMessage("initialSetupNotOnline"),
                 );
             }
         };
 
-        const showBuildOptions = function () {
-            const supported =
-                ((semver.eq(FC.CONFIG.apiVersion, API_VERSION_1_45) && ispConnected()) ||
+        // Fills in the "Firmware" part of the "Firmware info" box
+        const showBuildFirmware = function () {
+            const isIspConnected = ispConnected();
+            const buildOptionsValid =
+                ((semver.eq(FC.CONFIG.apiVersion, API_VERSION_1_45) && isIspConnected) ||
                     semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_46)) &&
                 FC.CONFIG.buildOptions.length;
-
-            if (supported) {
-                let buildOptionList = `<div class="dialogBuildInfoGrid-container">`;
-                for (const buildOptionElement of FC.CONFIG.buildOptions) {
-                    buildOptionList += `<div class="dialogBuildInfoGrid-item">${buildOptionElement}</div>`;
-                }
-                buildOptionList += `</div>`;
-
-                build_options_e.html(`<span class="buildInfoBtn" title="${i18n.getMessage(
```

Comment:
Note that in the original code the result of this call to `build_options_e.html(...)` was immediately overwritten by the following call to `build_options_e.html(buildOptions)`. I have deleted this statement as it seems unnecessary..